src/tls_stream.rs

```diff
@@ -501,6 +502,30 @@ impl<S> TlsStream<S>
         };
 
         unsafe {
+            // check if we trust the root-CA explicitly
+            let mut para_flags = winapi::CERT_CHAIN_POLICY_IGNORE_ALL_REV_UNKNOWN_FLAGS;
+            if let Some(ref mut store) = self.cert_store {
+                let cert_chain = *cert_chain.0;
+                if cert_chain.cChain > 0 {
+                    let first_rgp_chain = **cert_chain.rgpChain;
+                    if first_rgp_chain.cElement > 0 {
+                        let elements = slice::from_raw_parts(
```

Comment:
Did that in the first draft, personally I prefer this one, since there's less dereferencing.
and you'd need to do ```first_rpg_chain.cElement as isize - 1```

I think the current one is easier to get correct, so I'd prefer to keep it this way
(mainly because it just subjectively looks "nicer")?